homeassistant/util/unit_conversion.py

```diff
@@ -155,7 +155,11 @@ def converter_factory_allow_none(
             return lambda value: value
         from_ratio, to_ratio = cls._get_from_to_ratio(from_unit, to_unit)
         if cls._are_unit_inverses(from_unit, to_unit):
-            return lambda val: None if val is None else to_ratio / (val / from_ratio)
+            return (
+                lambda val: None
+                if val is None
+                else (0 if val == 0 else to_ratio / (val / from_ratio))
```

Comment:
Should this not return `None`?

Reply:
In my view, we have the following situation:

kWh/100km measures energy consumption (how much energy to travel a distance)
km/kWh measures efficiency (how far you can go on a unit of energy)

For any non-zero value, they're inversely related. If a car uses 15 kWh/100km, its efficiency is 100/15 ≈ 6.67 km/kWh.
But at zero, both statements mean the same thing: no energy is being used and/or no distance is being traveled. It's a degenerate case where the reciprocal relationship breaks down mathematically (you can't divide by zero), but conceptually "zero consumption" and "zero efficiency" both point to the same physical situation ... nothing is happening.